### ISSUE TITLE
WordPress.com Block Editor: wp.editor.RichTextToolbarButton is deprecated

### ISSUE THREAD

**Issue by jeherve** [Automattic/wp-calypso#55682]
The WordPress.com Block Editor pulls the `RichTextToolbarButton` element from `@wordpress/editor`:
https://github.com/Automattic/wp-calypso/blob/fb55dd173718474002a6204905974f2d9a669314/apps/wpcom-block-editor/src/default/features/rich-text.js#L5

However, that element should now be pulled from `@wordpress/block-editor` instead.

This consequently causes some `Deprecated wp.editor.RichTextToolbarButton` deprecation warnings when using the WordPress.com Block editor.

*******

Originally reported in https://github.com/Automattic/jetpack/issues/20806

**Comment by zachspears** [comment]
Hello @jeherve, thank you for sharing this issue. Could you please provide steps for reproducing the error? Thank you!

**Comment by cuemarie** [comment]
📌 SCRUBBING : RESULT 
- Tested on Self-hosted: Replicated

📌 FINDINGS/SCREENSHOTS/VIDEO
- Checked out the steps in this report: https://github.com/Automattic/jetpack/issues/20806  and took the following steps: 

1. Self-hosted site with Jetpack installed and activated
2. Edit a post
3. Open browser console
4. Select a Paragraph block

Doing so, I see this warning: 

```
`wp.data.select( 'core/editor' ).getSelectedBlock` is deprecated since version 5.3 and will be removed in version 6.2. Please use `wp.data.select( 'core/block-editor' ).getSelectedBlock` instead.
```

> This consequently causes some Deprecated wp.editor.RichTextToolbarButton deprecation warnings when using the WordPress.com Block editor

I can't speak to how to replicate where this originates, but I think this should be sufficient to replicate the result of the issue and move this forward. 

📌 ACTIONS
- Marked as Triaged for Quality Squad review

**Comment by agrullon95** [comment]
Warning @cuemarie noted above has been fixed with #69652 .

> `wp.data.select( 'core/editor' ).getSelectedBlock` is deprecated since version 5.3 and will be removed in version 6.2. Please use `wp.data.select( 'core/block-editor' ).getSelectedBlock` instead.
